features/step_definitions/minion_ssh_push_steps.rb

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2016 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+And(/^I enter the hostname of "([^"]*)" as hostname$/) do |minion|
+  if minion == "sle-minion"
+    code, _out = $server.run("grep \"#{$minion_hostname}\" /etc/hosts/", false)
+    $server.run("echo  \"#{$minion_ip}  #{$minion_hostname}\" >> /etc/hosts") if code != 0
+    step %(I enter "#{$minion_hostname}" as "hostname")
+
+  elsif minion == "rh-minion"
+    # THIS WILL NOT WORK UNTIL WE fully SUPPORT RH images.
+    code, _out = $server.run("grep #{$rh_minion_hostname} /etc/hosts/")
+    $server.run("echo  \"#{$rh_minion_ip}  #{$rh_minion_hostname}\" >> /etc/hosts") if code != 0
+    step %(I enter "#{$rh_minion_hostname}" as "hostname")
+  else
+    raise "no valid name of minion given! "
+  end
+end
+
+Given(/^the salt-master can reach "([^"]*)"$/) do |minion|
+  begin
+    start = Time.now
+    # 300 is the default 1st keepalive interval for the minion
+    # where it realizes the connection is stuck
+    Timeout.timeout(DEFAULT_TIMEOUT + 300) do
+      # only try 3 times
+      3.times do
+        if minion == "sle-minion"
+          _code, out = $server.run("salt #{$minion_hostname} test.ping", false)
+          if out.include?($minion_hostname)
+            finished = Time.now
+            puts "Took #{finished.to_i - start.to_i} seconds to contact the minion"
+            break
+          end
+
+        elsif minion == "rh-minion"
+          _code, out = $server.run("salt #{$rh_minion_hostname} test.ping", false)
+          if out.include?($minion_hostname)
+            finished = Time.now
+            puts "Took #{finished.to_i - start.to_i} seconds to contact the minion"
+            break
```

Comment:
This block could be refactored for 'sle-minion' and 'rh-minion'. What do you think?